app/js/components/manageApps/Addon.jsx

```diff
@@ -108,35 +106,59 @@ class Addon extends Component {
     );
   }
 
-  handleOwaDelete() {
+  handleAddonDelete() {
     event.preventDefault();
     this.setState({
       isOpen: true,
     });
   }
 
-  handleModuleDelete() {
-    event.preventDefault();
-  }
-
-  handleDelete(name) {
+  handleDelete(name, appUuid) {
     this.setState((prevState, props) => {
       return {
         deleting: true,
       };
     });
-    const applicationDistribution = location.href.split('/')[3];
-    axios.get(`/${applicationDistribution}/module/owa/deleteApp.form?appName=${name}`).then(response => {
-      this.setState((prevState, props) => {
-        return {
-          deleting: false,
-        };
+
+    if(appUuid != null){
+      const applicationDistribution = location.href.split('/')[2];
+      const url = location.href.split('/')[3];
+      const apiBaseUrl = `/${applicationDistribution}/${url}/ws/rest`;
+      this.requestUrl = '/v1/moduleaction';
+      let postData = {};
+
+      postData = {
+        "action": "unload",
+        "modules": [appUuid],
+      };
+
+      axios.post(`https:/${apiBaseUrl}${this.requestUrl}`, postData).then(response => {
```

Comment:
Nice implementation, leaves code cleaner and avoids those long lines. Kudos!

Reply:
Thanks